tests/RuleSet/RuleSetsTest.php

```diff
@@ -163,21 +163,62 @@ public function testPHPUnitMigrationTargetVersions(string $setName): void
         }
     }
 
-    public function testRegisterRuleSetOK(): void
+    /**
+     * @dataProvider provideRegisterRuleSetOkCases
+     */
+    public function testRegisterRuleSetOk(string $goodName): void
     {
-        RuleSets::registerRuleSet('@MyRules', SampleRulesOk::class);
-        self::assertContains('@MyRules', RuleSets::getSetDefinitionNames());
+        RuleSets::registerRuleSet($goodName, SampleRulesOk::class);
+        self::assertContains($goodName, RuleSets::getSetDefinitionNames());
+    }
+
+    /**
+     * @return array<int, array<int, string>>
+     */
+    public static function provideRegisterRuleSetOkCases(): array
+    {
+        $goodNames = [
+            '@MyRules',
+            '@MyRules-1.0',
+            '@MyRules_1_0',
+            '@MyRules/1.0',
+        ];
+
+        return array_map(static function (string $goodName): array {
+            return [$goodName];
+        }, $goodNames);
```

Comment:
```suggestion
    public static function provideRegisterRuleSetOkCases(): iterable
    {
        yield 'Simple name' => ['@MyRules'];
        yield 'Versioned with dash and X.Y' => ['@MyRules-1.0'];
        yield 'Versioned with underscores' => ['@MyRules_1_0'];
        yield 'Vendor/Ruleset' => ['@Vendor/MyRules'];
```

Please do something similar in `provideRegisterRuleSetBadNameCases()`.